client/src/app/shared/services/scenario/national-cloud.environment.ts

```
@@ -59,8 +60,11 @@ export class NationalCloudEnvironment extends AzureEnvironment {
 
         this.scenarioChecks[ScenarioIds.enableAppInsights] = {
             id: ScenarioIds.enableAppInsights,
-            runCheck: () => {
-                return { status: 'disabled' };
+            runCheckAsync: (input: ScenarioCheckInput) => {
```

>runCheckAsync [](start = 12, length = 13)

You don't need to implement runCheckAsync if you're not actually doing async work.  The ScenarioService is implemented so that if you call checkScenario and one environment has implemented an async check for a given id, it will throw and error.  So this change is fine, but the previous code was also fine.

---

If you feel strongly about this, I can change it back... but I like it this was because it maintain consistency across all files. like this if a change is made in the future, the dev does not have to try to figure out why one place it is runCheck but somewhere else its runCheckAsync..

---

Nope I don't feel strongly.  Your reasoning makes sense but I don't really expect people to have to go and change existing signatures if they're adding the first async check.  I'm not really worried since we fail fast so a developer will catch this pretty quickly if they're not calling the right checkScenario method.